### ISSUE TITLE
Select time zone for date module

### ISSUE THREAD

**Issue by nyarly** [jaagr/polybar#596]
It'd be extremely handy to be able to select which time zone a date module displays. Even just "utc = true" would be useful.

**Comment by jaagr** [comment]
See #557

**Comment by patrick96** [comment]
I reopened this because we will now start tracking this as a feature request for a new parameter in the date module and ignore the script workaround

**Comment by Lomadriel** [comment]
I tried something with `setenv` to set the `TZ` environment variable as suggested in #788 by @NBonaparte. But there is a problem.

If we suppose the following config:
```
[bar/example]
…
modules-left = date_france date_local
…
```
The environment modification applies to all threads, so if there is two `internal/date` modules, as in the example above, the timezone may be applied to the two modules rather than just the `date_france` module.

One solution could be to use a "global clock", each module will get the time from a blocking method that will use `setenv`. Since this is a blocking call the request from each module will be sequential and it will resolve the thread concurrency problem.
What do you think of this solution?

**Comment by patrick96** [comment]
This feature is currently blocked because this seems to be impossible to do safely. See [this comment](https://github.com/polybar/polybar/pull/3160#issuecomment-2323348167) for more details.

TL;DR using `setenv` is inherently unsafe and there is currently no other way to format a point in time for a specific timezone.
There is a C++20 feature that may allow us to do this safely, but C++20 is still experimental in GCC
